### PR TITLE
New features

### LINKED ISSUE #292: Import timetable
Since timetable is stored in cache, if its get erased and we have to make new changes in the timetable then we have to make the timetable again. A similar option to the export timetable option, could be made which accepts file as ics format and renders it on webpage.

### LINKED ISSUE #341: Add Undo
**Describe the solution you'd like**
The ability to undo changes

**Describe Possible Solutions**
Maybe a shortcut (ctrl z) and a button in nav bar

**Additional context**
N/A

### LINKED ISSUE #343: Export to link
**Describe the solution you'd like**
Share/save timetables using a link.

### PR DESCRIPTION
**Pull request by ggggg**
## feat/[name]

Fixes: #341 #343

- [ ] Has the code been formatted?
- [ ] Have dependencies/packages been added/removed?
      If so, please list them.

### Changes

List out all the changes, and then explain the changes made at a higher level

- Added undo and redo buttons
- Added export and import buttons that export the current timetable to a JSON file.

### CODE REVIEW

**Review by logonoff — APPROVED** [review]

File: src/components/FloatingBtns/ExportBtn.vue
Comment (huonggiangbui) [inline src/components/FloatingBtns/ExportBtn.vue:1]: Prolly rename this module. I guess ExportBtn is to export as JSON, where ImageBtn is to export as PNG. Needs a more readable name for them

File: src/store/index.js [hunk 1]
```diff
@@ -340,8 +370,33 @@ export default new Vuex.Store({
     setPreferredDeliveryMethod(state, payload) {
       state.deliveryMethod = payload;
     },
-    setGlobalAllowConflicts(state, payload) {
-      state.globalAllowConflicts = payload;
+    setGlobalAllowConflicts(state, payload)
```
Comment (huonggiangbui) [inline src/store/index.js]: Don't quite get this changes. Can you explain this?

File: src/views/Home.vue
Comment (huonggiangbui) [inline src/views/Home.vue:1]: Don't know if undo/redo feature is a good thing to add. Need to discuss with the team

**Review by huonggiangbui — COMMENTED** [review]
Leave some comments for discussion before merging.

Also, if you can add those features to our changelog/documentation, and the welcome dashboard, that would be great!

Thank you for this hard work !!

File: src/store/index.js (hunk 1 above)
Comment (ggggg) [reply src/store/index.js]: Seems to just be a minor formatting change, not a major change. I can change it back if you want

File: src/views/Home.vue
Comment (ggggg) [reply src/views/Home.vue:1]: https://discord.com/channels/840580050632835072/858685332203896834/1118032263163166770

File: src/components/FloatingBtns/ExportBtn.vue
Comment (ggggg) [reply src/components/FloatingBtns/ExportBtn.vue:1]: Done f62e4b3b

**Review by logonoff — APPROVED** [review]
lgtm!

**Pull request merged**